realkd/utils.py

```diff
@@ -0,0 +1,166 @@
+        >>> a = RealkdArrayLike(data)
+        >>> a[1]
+        RealkdArrayLike<_raw=[2 8], labels=['x0', 'x1']>
+        >>> a[:, 1]
+        RealkdArrayLike<_raw=[7 8 9], labels=['x0', 'x1']>
+        >>> b = RealkdArrayLike(pd.DataFrame(data, columns=a.labels))
+        >>> b[1]
+        RealkdArrayLike<_raw=x0    2
+        x1    8
+        Name: 1, dtype: int64, labels=['x0', 'x1']>
+        >>> b[:, 1]
+        RealkdArrayLike<_raw=0    7
+        1    8
+        2    9
+        Name: x1, dtype: int64, labels=['x0', 'x1']>
+        >>> a['x1'][1] + 1
+        9
+        >>> b['x1'][1] + 1
+        9
+        >>> len(a['x1'])
+        3
+        >>> c = RealkdArrayLike(np.array([[1., np.NaN],[2, 8],[3, 9]]))
+        >>> col = c[:, 1]
+        >>> col[~pd.isnull(col)]
+        RealkdArrayLike<_raw=[8. 9.], labels=['x0', 'x1']>
+        """
+        data, labels = self._index(keys)
+        if hasattr(data, "__len__"):
+            return RealkdArrayLike(data, labels)
+        return data
+
+    def _index(self, first_key, *more_keys):
+        if type(first_key) == str:
+            if hasattr(self._raw, "iloc"):
+                return self._raw.__getitem__(first_key), self.labels
+            if hasattr(self._raw, 'shape'):
+                # If indexing a numpy array with a string, the logic is:
+                # If the array is 1D, treat it as an observation
+                # If the array is 2D, treat it the first dimention (rows) as observations
+                if self._raw.ndim == 1:
+                    return self._raw[self.labels.index(first_key)], [first_key]
+                return self._raw[:, self.labels.index(first_key)], [first_key]
+            return self._raw.__getitem__(first_key), [first_key]
+
+        else:
+            if hasattr(self._raw, "iloc"):
+                try:
+                    return self._raw.iloc.__getitem__(first_key, *more_keys), self.labels
+                except:
+                    return self._raw.loc.__getitem__(first_key, *more_keys), self.labels
+            return self._raw.__getitem__(first_key, *more_keys), self.labels
+
+    def __len__(self):
+        return len(self._raw)
+
+    def __eq__(self, other):
+        """
+        >>> c = RealkdArrayLike(np.array([[1, 5],[2, 8],[3, 9]]))
+        >>> 2 < c
+        array([[False,  True],
+               [False,  True],
+               [ True,  True]])
+        >>> d = RealkdArrayLike(pd.DataFrame(np.array([[1, 5],[2, 8],[3, 9]])))
+        >>> d > 2
+               0     1
+        0  False  True
+        1  False  True
+        2   True  True
+        """
+        return self._raw.__eq__(other)
+
+    def __gt__(self, other): return self._raw.__gt__(other)
+    def __ge__(self, other): return self._raw.__ge__(other)
+    def __lt__(self, other): return self._raw.__lt__(other)
+    def __le__(self, other): return self._raw.__le__(other)
+    def __neg__(self): return self._raw.__neg__()
+
+    def __getattr__(self, name):
```

Comment:
Do you think this is a good idea or is it **too** permissive?

I thought a potential middle ground could be to add a 
print("Warning - this may be 100% supported") before attribute access so that our own code doesn't accidentally use it and couple itself to a particular data structure? What do you think?